File: 3096/main.go
package main

import "fmt"

func main() {
	fmt.Println(minimumLevels([]int{1,0,1,0}))
}

func minimumLevels(possible []int) int {
	n := len(possible)
	tot := 0
	for _, v := range possible {
		tot += v
	}
	tot = tot * 2 - n
	pre := 0
	for i := 0; i < n - 1; i++ {
		if possible[i] == 1 {
			pre++
		} else {
			pre--
		}
		if 2 * pre > tot {
			return i + 1
		}
	}
	return -1
}

File: 3096/main.rs
struct Solution{}

impl Solution {
    pub fn minimum_levels(possible: Vec<i32>) -> i32 {
        let n = possible.len() as i32;
        let mut tot = 0;
        for v in possible.iter() {
            tot += v;
        }
        tot = tot * 2 - n;
        let mut pre = 0;
        for i in 0..n-1 {
            if possible[i as usize] == 1 {
                pre += 1;
            } else {
                pre -= 1;
            }
            if 2 * pre > tot {
                return i + 1
            }
        }
        -1
    }
}

fn main() {
    println!("{}", Solution::minimum_levels(vec![1,0,1,0]))
}